tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -804,3 +806,445 @@ TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST)
+                    DataEnvironment::DayOfWeek = 1;
+                }
+                ++DataEnvironment::DayOfWeekTomorrow;
+                if (DataEnvironment::DayOfWeekTomorrow > 7) {
+                    DataEnvironment::DayOfWeekTomorrow = 1;
+                }
+
+                DataEnvironment::DayOfYear_Schedule = General::OrdinalDay(DataEnvironment::Month, DataEnvironment::DayOfMonth, 1);
+
+                for (int hr = 1; hr <= 24; ++hr) {
+                    ++HourOfYear;
+                    DataGlobals::HourOfDay = hr;
+                    for (int ts = 1; ts <= 4; ++ts) {
+                        DataGlobals::TimeStep = ts;
+
+                        ScheduleManager::UpdateScheduleValues();
+                        EXPECT_EQ(HourOfYear, ScheduleManager::LookUpScheduleValue(1, DataGlobals::HourOfDay, DataGlobals::TimeStep));
+                        EXPECT_EQ(HourOfYear, ScheduleManager::Schedule(1).CurrentValue);
+                        EXPECT_EQ(HourOfYear, ScheduleManager::GetCurrentScheduleValue(1));
+                    }
+                }
+            }
+        }
+
+        EXPECT_EQ(8760.0, HourOfYear);
+    }
+
+    {
+        DataEnvironment::DSTIndicator = 1; // DST IS ON
+        DataEnvironment::DayOfWeek = 0;
+        DataEnvironment::DayOfWeekTomorrow = 1;
+
+        Real64 HourOfYear = 0.0;
+        for (int month=1; month <= 12; ++month) {
+            DataEnvironment::Month = month;
+            for (int day = 1; day <= EndDayOfMonth(month); ++day) {
+                DataEnvironment::DayOfMonth = day;
+                ++DataEnvironment::DayOfWeek;
+                if (DataEnvironment::DayOfWeek > 7) {
+                    DataEnvironment::DayOfWeek = 1;
+                }
+                ++DataEnvironment::DayOfWeekTomorrow;
+                if (DataEnvironment::DayOfWeekTomorrow > 7) {
+                    DataEnvironment::DayOfWeekTomorrow = 1;
+                }
+
+                DataEnvironment::DayOfYear_Schedule = General::OrdinalDay(DataEnvironment::Month, DataEnvironment::DayOfMonth, 1);
+
+                for (int hr = 1; hr <= 24; ++hr) {
+                    ++HourOfYear;
+                    DataGlobals::HourOfDay = hr;
+                    for (int ts = 1; ts <= 4; ++ts) {
+                        DataGlobals::TimeStep = ts;
+
+                        ScheduleManager::UpdateScheduleValues();
+                        int thisHourOfYear = HourOfYear + 1;
+                        if (thisHourOfYear > 8760.0) {
+                            thisHourOfYear = 1;
+                        }
+                        EXPECT_EQ(thisHourOfYear, ScheduleManager::LookUpScheduleValue(1, DataGlobals::HourOfDay, DataGlobals::TimeStep));
+                        EXPECT_EQ(thisHourOfYear, ScheduleManager::Schedule(1).CurrentValue);
+                        EXPECT_EQ(thisHourOfYear, ScheduleManager::GetCurrentScheduleValue(1));
+                    }
+                }
```

Comment:
Same for the DST case, checking the special case at the last timestep which should be January 1 as a result (value == 1)